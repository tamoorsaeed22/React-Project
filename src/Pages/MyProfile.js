import { useSelector } from 'react-redux';
import Header from '../Components/Header';
import styles from './MyProfile.module.css';
import { selectRockets } from '../app/features/rocketsReducer';
import { selectMissions } from '../app/features/missionReducer';

const MyProfile = () => {
  const rocketsList = useSelector(selectRockets);
  const missionsList = useSelector(selectMissions);
  const reservedRockets = rocketsList.filter((b) => b.reserved);
  const joinedMissions = missionsList.filter((b) => b.joined);
  const generateKey = () => `_${Math.random().toString(36).substr(2, 9)}`;
  return (
    <>
      <Header />
      <section className={styles.myProfileSection}>
        <div className={styles.myProfileMissionsList}>
          <h1>My Missions</h1>
          <ul className={styles.myProfileMissionsList}>
            {joinedMissions.map((mission) => (
              <li key={generateKey()}>{mission.mission_name}</li>
            ))}
          </ul>
        </div>
        <div className={styles.myProfileRocketsList}>
          <h1>My Rockets</h1>
          <ul className={styles.myProfileRocketsList}>
            {reservedRockets.map((rocket) => (
              <li key={generateKey()}>{rocket.rocket_name}</li>
            ))}
          </ul>
        </div>
      </section>
    </>
  );
};

export default MyProfile;
