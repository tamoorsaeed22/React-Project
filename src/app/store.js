import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './features/rocketsReducer';
import missionsReducer from './features/missionReducer';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
