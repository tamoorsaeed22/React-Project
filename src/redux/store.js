import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rocket/rocketsReducer';
import missionsReducer from './mission/missionReducer';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
