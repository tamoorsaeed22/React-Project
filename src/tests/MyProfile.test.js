import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import MyProfile from '../Pages/MyProfile';

describe('MyProfile Test', () => {
  it('renders correctly', () => {
    const myProfileTest = renderer.create(
      <Provider store={store}>
        <Router>
          <MyProfile />
        </Router>
      </Provider>,
    ).toJSON();
    expect(myProfileTest).toMatchSnapshot();
  });
});
