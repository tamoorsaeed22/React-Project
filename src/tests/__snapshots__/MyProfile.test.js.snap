// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Test renders correctly 1`] = `
Array [
  <header>
    <div
      className="header-logo-and-name"
    >
      <img
        alt="planet logo"
        className="header-logo"
        src="planet.png"
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </div>
    <nav>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "bold",
            "textDecoration": "underline",
          }
        }
      >
        Rockets
      </a>
      <a
        className=""
        href="/Missions"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "normal",
            "textDecoration": "none",
          }
        }
      >
        Missions
      </a>
      <a
        className="profile"
        href="/MyProfile"
        onClick={[Function]}
        style={
          Object {
            "fontWeight": "normal",
            "textDecoration": "none",
          }
        }
      >
        My Profile
      </a>
    </nav>
  </header>,
  <section
    className="myProfileSection"
  >
    <div
      className="myProfileMissionsList"
    >
      <h1>
        My Missions
      </h1>
      <ul
        className="myProfileMissionsList"
      />
    </div>
    <div
      className="myProfileRocketsList"
    >
      <h1>
        My Rockets
      </h1>
      <ul
        className="myProfileRocketsList"
      />
    </div>
  </section>,
]
`;
